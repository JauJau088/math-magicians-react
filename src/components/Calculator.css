.CalculatorContainer {
  display: flex;
  flex-direction: column;
  width: 330px;
}

.Display {
  max-width: 100%;
  height: 45px;
  padding: 5px 10px;
  background-color: #858694;
  color: white;
  font-size: 1.2rem;
}

.Display-Text {
  display: flex;
  justify-content: end;
  align-items: center;
  text-align: end;
  width: 100%;
  height: 100%;
}

.ButtonsContainer {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  background-color: grey;
  gap: 1px;
  padding: 1px;
}

.Button {
  min-width: 80px;
  max-width: 100%;
  height: 45px;
  border: none;
}

.Grey {
  background-color: #e0e0e0;
}

.Orange {
  background-color: #f5913e;
}

.ColTwo {
  grid-area: auto / span 2;
}
