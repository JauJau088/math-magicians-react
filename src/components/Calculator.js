import React, { useState } from 'react';
import './Calculator.css';
import Button from './CalculatorButton';
import calculate from '../logic/calculate';

const Calculator = () => {
  // State hook
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  // Calc function
  const calc = (btnName) => {
    setState(calculate(state, btnName));
  };

  // Display logic
  const { total, next, operation } = state;
  const subShow = operation == null ? '' : `${total} ${operation}`;
  const show = next || total;

  // Render
  return (
    <div className="SectionContainer SectionCalculator">
      <div className="ChalkBoard LetsMath">
        <div>Let&apos;s do some math!</div>
      </div>
      <div className="CalculatorContainer">
        <h1 className="logo">JauJau</h1>
        <div className="Display">
          <div className="Display-SubText">{subShow}</div>
          <div className="Display-Text">{show}</div>
        </div>
        <hr />
        <div className="ButtonsContainer">
          <Button text="AC" callback={calc} />
          <Button text="+/-" callback={calc} />
          <Button text="%" callback={calc} />
          <Button text="รท" callback={calc} bgColor="Orange" />
          <Button text="7" callback={calc} />
          <Button text="8" callback={calc} />
          <Button text="9" callback={calc} />
          <Button text="x" callback={calc} bgColor="Orange" />
          <Button text="4" callback={calc} />
          <Button text="5" callback={calc} />
          <Button text="6" callback={calc} />
          <Button text="-" callback={calc} bgColor="Orange" />
          <Button text="1" callback={calc} />
          <Button text="2" callback={calc} />
          <Button text="3" callback={calc} />
          <Button text="+" callback={calc} bgColor="Orange" />
          <Button text="0" callback={calc} col="ColTwo" />
          <Button text="." callback={calc} />
          <Button text="=" callback={calc} bgColor="Orange" />
        </div>
      </div>
    </div>
  );
};

export default Calculator;
