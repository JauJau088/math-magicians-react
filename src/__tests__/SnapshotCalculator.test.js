import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('Snapshot test for Calculator page', () => {
  it('It should render an identical template', () => {
    const CalculatorPage = renderer.create(<Calculator />).toJSON();
    expect(CalculatorPage).toMatchSnapshot();
  });
});
