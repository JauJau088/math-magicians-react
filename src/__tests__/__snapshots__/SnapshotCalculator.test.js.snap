// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Calculator page It should render an identical template 1`] = `
<div
  className="SectionContainer SectionCalculator"
>
  <div
    className="ChalkBoard LetsMath"
  >
    <div>
      Let's do some math!
    </div>
  </div>
  <div
    className="CalculatorContainer"
  >
    <h1
      className="logo"
    >
      JauJau
    </h1>
    <div
      className="Display"
    >
      <div
        className="Display-SubText"
      />
      <div
        className="Display-Text"
      />
    </div>
    <hr />
    <div
      className="ButtonsContainer"
    >
      <button
        className="Button Grey "
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="Button Grey "
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="Button Grey "
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="Button Orange "
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="Button Grey "
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="Button Grey "
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="Button Grey "
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="Button Orange "
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="Button Grey "
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="Button Grey "
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="Button Grey "
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="Button Orange "
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="Button Grey "
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="Button Grey "
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="Button Grey "
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="Button Orange "
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="Button Grey ColTwo"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="Button Grey "
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="Button Orange "
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
